### PR TITLE
Improve Docker build

### LINKED ISSUE #158: Docker entrypoint does not pass on command line arguments
The Docker entrypoint does not pass command line arguments that are given to `docker run`:

https://github.com/google/copybara/blob/a66f59c5898cda9a3a20ffceecfad1d0f506e456/.docker/entrypoint.sh#L17

This presents a problem for running workflows that accept command line arguments. 

Further, I'm unclear why `COPYBARA_*` env vars were preferred over simply doing something like:

```
java -jar /opt/copybara/copybara_deploy.jar "$@"
```

Any pointers much appreciated.

Thanks

### LINKED ISSUE #235: Docker entrypoint script fails when no COPYBARA_OPTIONS flag passed
Ran into this during deployment using a dockerized implementation:

```
+ java -jar /opt/copybara/copybara_deploy.jar '' migrate copy.bara.sky push_examples_private_to_public ''
May 16, 2023 6:42:59 PM com.google.copybara.Main configureLog
INFO: Setting up LogManager
Copybara (Version: Unknown version)
0516 18:43:00.214 ERROR: Invalid subcommand ''. Available commands: [generator, help, info, migrate, onboard, validate, version]
0516 18:43:00.217 ERROR: Try 'copybara help'.
```

Because of the ordering in the `[entrypoint.sh](https://github.com/google/copybara/blob/47ce5ccd468b46bf3a813a0ce3e83ee44fead192/.docker/entrypoint.sh#L17)` script, it assumes there's some value to be passed for the COPYBARA_OPTIONS. Moving COPYBARA_SUBCOMMAND up the order in that script resolves the issue. 

Note that we opted to use an ENTRYPOINT in the Dockerfile, so we're deviating a little bit from the existing Dockerfile, but because we wanted to abstract away all the command syntax from our devs and just expose the config options as envvars.

### PR DESCRIPTION
**Pull request by mering**
Fixes #158
Fixes #235

### CODE REVIEW

**Review by mikelalcon — COMMENTED** [review]
I don't know much about docker, but the deletion of this seems that would break existing users:

COPYBARA_CONFIG=copy.bara.sky`
  * allows you to specify a path to a config file, defaults to root `copy.bara.sky`
* `COPYBARA_SUBCOMMAND=migrate`
  * allows you to change the command run, defaults to `migrate`
* `COPYBARA_OPTIONS=''`
  * allows you to specify options for copybara, defaults to none
* `COPYBARA_WORKFLOW=default`
  * allows you to specify the workflow to run, defaults to `default`
* `COPYBARA_SOURCEREF=''`

**Comment by mering** [comment]
If you want to handle environment arguments, this should definitely done by the executable itself. Adding this as a script to the container, feels like a big hack.

Nevertheless, I added that functionality back to not break existing users for the time being.

**Review by mikelalcon — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -13,29 +13,21 @@
...
-WORKDIR /usr/src/app
-ENV COPYBARA_CONFIG=copy.bara.sky \
-    COPYBARA_SUBCOMMAND=migrate \
-    COPYBARA_OPTIONS='' \
-    COPYBARA_WORKFLOW=default \
-    COPYBARA_SOURCEREF=''
-COPY --from=build /tmp/copybara/ /opt/copybara/
+RUN apt-get update && apt-get install -y \
+        git \
+    && rm -rf /var/lib/apt/lists/*
```
[the remark this answers is not in the document]
Comment (mering) [reply Dockerfile:26]: This is suggested as best practice from Docker itself: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get

This is basically what is used in more than 99% of all `Dockerfile`s. Should I still add a comment?

[the remark this answers is not in the document]
Comment (mikelalcon) [reply Dockerfile:26]: Yes, at least in the commit description. Can you: squash all changes in one and add a commit description that explains all we are doing here?

Thanks!

[the remark this answers is not in the document]
Comment (mering) [reply Dockerfile:27]: Done

File: Dockerfile [hunk 2]
```diff
@@ -13,29 +13,21 @@
...
+COPY --from=build /home/ubuntu/bazel-bin/java/com/google/copybara/copybara_deploy.jar /opt/copybara/
 COPY --from=buildtools /go/bin/buildozer /go/bin/buildifier /usr/bin/
-COPY .docker/entrypoint.sh /usr/local/bin/copybara
-RUN chmod +x /usr/local/bin/copybara
-RUN apt-get update && \
-    apt-get install -y git && \
-    apt-get clean
+# NOTE NullPointerException when using directly ENTRYPOINT["java","-jar","/opt/copybara/copybara_deploy.jar"]
```
[the remark this answers is not in the document]
Comment (mering) [reply Dockerfile]: Fixed in #255

**Pull request closed without merging**